tour_offers/src/main/kotlin/pl/edu/pg/rsww/touroffers/TourOffersManager.kt

```diff
@@ -0,0 +1,126 @@
+package pl.edu.pg.rsww.touroffers
+
+import com.mongodb.kotlin.client.MongoClient
+import com.mongodb.kotlin.client.MongoDatabase
+import com.mongodb.client.model.Filters
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.bson.Document
+
+data class Entity(val entity_id: String, val entity_type: String, val last_event_id: Int, val data: Document)
+
+public class TourOffersManager{
+    val pageSize: Int = 5
+    val userName = System.getenv("MONGO_USERNAME")
+    val password = System.getenv("MONGO_PASSWORD")
+    val dbName = System.getenv("MONGO_DB")
+    val host = System.getenv("MONGO_HOSTB")
+    val port = System.getenv("MONGO_PORTB")
+    val connectionString = "mongodb://$userName:$password@$host:$port/"
+
+    /*fun GetAvailable(db: MongoDatabase): List<String> {
+        val projection = Projection.fields(
+                Projections.include(Entity::entity_id.name),
+                Projection.excludeId()
+        )
+
+        return db.getCollection<Entity>("snapshots").find(Filters.eq(Entity::entity_type.name, "Hotel"))
+                .toList()
+                .filter{
+                    it.data?.getInteger("reservation_count") <= it.data?.getInteger("reservation_limit") ?: false
+                }
+    }*/
+
+    fun getTourList(destination: String,
+                    from: String,
+                    departureDate: String,
+                    adults: Int,
+                    under3: Int,
+                    under10: Int,
+                    under18: Int): String{
+
+        val client = MongoClient.create(connectionString = connectionString)
+        val db = client.getDatabase(databaseName = dbName)
+
+        var minAge: Int = 18
+        val numPeople: Int = adults + under3 + under10 + under18
+        if (under3 != 0){
+            minAge = 0
+        }else if(under10 != 0){
+            minAge = 3
+        }else if(under18 != 0){
+            minAge = 10
+        }
+
+        var result = ""
+
+        val destCities = db.getCollection<Entity>("snapshots").find(Filters.eq(Entity::entity_type.name, "Tour"))

```

Comment:
The name of the variable is inaccurate considering what it actually contains, maybe `availableTours` would be better?
```suggestion
        val availableTours = db.getCollection<Entity>("snapshots").find(Filters.eq(Entity::entity_type.name, "Tour"))

```